JIT compile error with OpenJDK method handles

See https://openj9-jenkins.osuosl.org/job/Build_JDK8_x86-64_windows_ojdk292_Personal/1/ for this:
```
13:09:40  F:\Users\jenkins\workspace\Build_JDK8_x86-64_windows_ojdk292_Personal\openj9\runtime\compiler\optimizer\J9RecognizedCallTransformer.cpp(547) : error C3861: 'snprintf': identifier not found
13:09:40  make[4]: *** [runtime/compiler/CMakeFiles/j9jit.dir/build.make:2622: runtime/compiler/CMakeFiles/j9jit.dir/optimizer/J9RecognizedCallTransformer.cpp.obj] Error 2
```

--- a/runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp
+++ b/runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp
@@ -37,6 +37,7 @@
 #include "ilgen/IlGenRequest.hpp"
 #include "ilgen/IlGeneratorMethodDetails.hpp"
 #include "ilgen/IlGeneratorMethodDetails_inlines.hpp"
+#include "infra/String.hpp"
 #include "optimizer/CallInfo.hpp"
 #include "optimizer/IdiomRecognitionUtils.hpp"
 #include "optimizer/Structure.hpp"
@@ -544,7 +545,7 @@ getSignatureForComputedCall(
    const char * const returnTypeEnd = nextSignatureArgument(returnType);
 
    // Put together the new signature.
-   snprintf(
+   TR::snprintfNoTrunc(
       computedCallSignature,
       signatureAllocSize,
       "(%s%.*s%s)%.*s",

